addons/mail/models/mail_thread.py

```diff
@@ -2135,6 +2135,41 @@ def _message_create(self, values_list):
     # NOTIFICATION API
     # ------------------------------------------------------
 
+    @api.model
+    def discard_notifications(self, notification_type):
```

Comment:
I would add an explicit check this is called only by internal users. Previously wizard was limited to those people (see acls using base.group_user). I would therefore do
```
def notify_cancel_by_type(self, notification_type):
    if not self.env.user.has_group('base.group_user'):
        raise AccessError(somemessage)
```
We could maybe check that current user can read current model just to add some safety, like
```
    # limit to models current user has any chance to read
    self.check_access_rights('read')
```